prometheus/exporter/intervalpush/exporter.go

```diff
@@ -39,6 +40,59 @@ func NewExporter(url string, interval time.Duration) exp.Exporter {
 	}
 }
 
+type Option func(exporter *exporter)
+
+func WithInterval(interval time.Duration) Option {
+	return func(e *exporter) {
+		e.td = interval
+	}
+}
+
+func WithEnvironment(env string) Option {
+	return func(e *exporter) {
+		e.p = e.p.Grouping("env", env)
+	}
+}
+
+func WithAppName(name string) Option {
+	return func(e *exporter) {
+		e.p = e.p.Grouping("app", name)
+	}
+}
+
+func WithProject(project string) Option {
+	return func(e *exporter) {
+		e.p = e.p.Grouping("project", project)
+	}
+}
+
+func WithGrouping(name, value string) Option {
+	return func(e *exporter) {
+		e.p = e.p.Grouping(name, value)
+	}
+}
+
+func NewExporterWithOptions(url string, opts ...Option) exp.Exporter {
+	var e = exporter{
+		Monitor: closer.NewMonitor(closer.WithClosingPolicy(closer.Wait)),
+		td:      defaultInterval,
+		p: push.New(
+			url,
+			os.Getenv("APP_NAME"),
+		).Gatherer(
+			prometheus.DefaultGatherer,
+		),
+	}
+
+	for _, opt := range opts {
+		opt(&e)
+	}
+
+	e.t = time.NewTicker(e.td)
```

Comment:
```suggestion
	e.t = time.NewTicker(e.td)
```

You could move this into the structure initialization, it doesn't change much as options are non blocking and nobody is yet consuming the ticker channel.